src/Controller/RegistrationController.php

```diff
@@ -24,12 +24,7 @@ public function register(Request $request, UserService $userService): Response
         // 2) handle the submit (will only happen on POST)
         $form->handleRequest($request);
         if ($form->isSubmitted() && $form->isValid()) {
-
-            $password = $userService->encodePassword($user);
-            $user->setPassword($password);
-
-            $userService->persist($user);
-
+            $user = $userService->newUser($user);
```

Comment:
do you need the user here?

Reply:
I might... I know I already have User object accessible in this point of code but I thought it is better that newUser() returns User, and so this variable is handling that return value.

Reply:
`I might` but you don't at least not know, so no point in having this. [YAGNI](https://martinfowler.com/bliki/Yagni.html)

Reply:
So it is ok not to catch return value?
Also, when I redirect user to actual page where he is logged in(showing pets or whtevr), I will need $user object to pass it's name(for example) to redirectToRoute()

Reply:
I am saying you don't even have to return it from `newUser` and you don't have to cache it. I am just saying don't do stuff you don't need currently, only the things you need. A lot of stuff you think you might need ends up removed as it was never needed.